rubin_scheduler/scheduler/schedulers/queue_manager.py

```diff
@@ -0,0 +1,134 @@
+
+    def set_queue(self, observation_array):
+        self.desired_observations_array = observation_array
+        self.need_observing = np.ones(observation_array.size, dtype=bool)
+
+    def add_observation(self, observation, **kwargs):
+        if self.desired_observations_array is not None:
+            match_indx = np.where(self.desired_observations_array["target_id"] == observation["target_id"])
+            self.need_observing[match_indx] = False
+        for bf in self.basis_functions:
+            bf.add_observation(observation, **kwargs)
+        for detailer in self.detailers:
+            detailer.add_observation(observation, **kwargs)
+
+    def add_observations_array(self, observation_array, observations_hpid_in):
+        if self.desired_observations_array is not None:
+            indx = np.isin(self.desired_observations_array["target_id"], observation_array["target_id"])
+            self.need_observing[indx] = False
+
+            good = np.isin(observations_hpid_in["ID"], observation_array["ID"])
+            observations_hpid = observations_hpid_in[good]
+            for bf in self.basis_functions:
+                bf.add_observations_array(observation_array, observations_hpid)
+            for detailer in self.detailers:
+                detailer.add_observations_array(observation_array, observations_hpid)
+
+    def compute_reward(self, conditions):
+        reward = 0
+        for bf in self.basis_functions:
+            reward += bf(conditions)
+        return reward
+
+    def find_valid_observations_indx(self, conditions):
+        """Return the indices of desired_observations_array
+        that can be observed
+        """
+        valid = np.where(self.need_observing)[0]
+
+        # Compute reward from basis functions
+        reward = self.compute_reward(conditions)
+        # If reward is an array, then it's a HEALpy map and we
+        # need to interpolate to the actual positions we want.
+        # now to interpolate to the reward positions
+        if np.size(reward) > 1:
+            reward_interp = hp.get_interp_val(
+                reward,
+                np.degrees(self.desired_observations_array["RA"][valid]),
+                np.degrees(self.desired_observations_array["dec"][valid]),
+                lonlat=True,
+            )
+            sub_valid_reward = np.isfinite(reward_interp)
+            valid = np.ravel(valid[sub_valid_reward])
+
+        return valid
+
+    def request_observation(self, conditions, whole_queue=False, n_return=1):
+        """
+        Parameters
+        ----------
+        conditions : ``
+            Conditions object
+        whole_queue : `bool`
+            If True, return all desired observations with no check if
+            they are accessible.
+        n_return : `int`
+            Number of observations to return.
+        """
+
+        # Everything that needs observing
+        if whole_queue:
+            indx = np.where(self.need_observing)[0]
```

Comment:
Why wouldn't you check the conditions + basis functions too, for the whole queue?